nucliadb_utils/src/nucliadb_utils/audit/stream.py

```diff
@@ -54,6 +59,78 @@
 KB_USAGE_STREAM_AUDIT = "kb-usage.nuclia_db"
 
 
+class RequestContext:
+    def __init__(self):
+        self.audit_request: AuditRequest = AuditRequest()
+        self.start_time: float = time.monotonic()
+
+
+request_context_var = contextvars.ContextVar[Optional[RequestContext]]("request_context", default=None)
+
+
+def get_trace_id() -> str:
+    span = get_current_span()
+    if span is None:
+        return ""
+    return format_trace_id(span.get_span_context().trace_id)
+
+
+def get_request_context() -> Optional[RequestContext]:
+    return request_context_var.get()
+
+
+class AuditMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
+        context = RequestContext()
+        token = request_context_var.set(context)
+        context.audit_request.time.FromDatetime(datetime.now(tz=timezone.utc))
+        response = await call_next(request)
```

Comment:
are unhandled error already handled once we hit this line of code? or should there be a try/catch here?

Reply:
already handled in the same way of any other unhander error anywhere else